### PR TITLE
Net pricing in CHM API

### PR DESCRIPTION
**Pull request by patrikmiksovsky95**

### CODE REVIEW

File: mews-api.md [hunk 1]
```diff
@@ -563,9 +563,8 @@ There are certain rules that need to be followed in order for Mews to process th
   * The whole group is uniquely identified by `channelManagerId` in Mews and in the channel manager extranet.
   * Each reservation should have a unique code within the group. The same code for the reservation should be provided in any following modification message.
   * Each reservation in the group can have different `start`, `end`, `spaceTypeCode`, `ratePlanCode`.
-* Group total cost `group`.`totalCost` is the sum of each `reservation`.`totalCost`, which is the sum of all night costs and total costs of all `extras` of the `reservation`.
-  * All costs in the message are inclusive of VAT.
-  * If for any reason the `group`.`totalCost` is be different, Mews will automatically distribute the missing/additional cost to the nights, so the `group`.`totalCost` is achieved.
+* Group total cost `group`.`totalAmount` is the sum of each `reservation`.`totalAmount`, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.
+  * If for any reason the `group`.`totalAmount` is be different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.
```
Comment (Ksenia-Galtsova) [inline mews-api.md]: is different

Comment (Ksenia-Galtsova) [reply mews-api.md]: To add one more line: 
A reservation group `amount` should be either gross or net. Both `grooss` and `net` amounts, or one of them should be sent.

File: mews-api.md [hunk 2]
```diff
@@ -641,6 +640,11 @@ There are certain rules that need to be followed in order for Mews to process th
         {
           "code": "1",
           "cost": 20,
+          "amount": 
+          {
```
Comment (vitasamek) [inline mews-api.md]: a brace should be on the previous line

File: mews-api.md [hunk 3]
```diff
@@ -701,7 +734,12 @@ There are certain rules that need to be followed in order for Mews to process th
       "to": "2020-05-09"
     }
   ],
-  "totalCost": 600
+  "totalCost": 600,
+  "totalAmount": 
+  {
```
Comment (vitasamek) [inline mews-api.md]: a brace should be on the previous line

File: mews-api.md [hunk 4]
```diff
@@ -820,6 +858,14 @@ _¹ It is required that the code remains the same within each booking modificati
...
 | `pricing` | `int` | required | [`Extra pricing Type`](mews-api.md#extra-pricing-types) code of the extra product pricing. |
 
+#### Amount
+
+| Property | Type |  | Description |
+| --- | --- | --- | --- |
+| `net` | `decimal` | optional | Price with taxes excluded. |
+| `gross` | `decimal` | optional | Price with taxes included. |
+
+
```
Comment (vitasamek) [inline mews-api.md]: Remove the empty line

**Comment by Ksenia-Galtsova** [comment]
We should add description to **Extra** in processGroup to make it clear what amounts should be sent. Smth like this:
Prices should be sent in `net` or `gross` amounts. Both `grooss` and `net` amounts, or one of them should be sent. @patrikmiksovsky95

**Comment by siroky** [comment]
@patrikmiksovsky95 this is stale for long time, PING.
@patrikmiksovsky95 consider weekly check using pull panda to identify cases like this in your team.

File: mews-api.md [hunk 5]
```diff
@@ -641,6 +641,11 @@ There are certain rules that need to be followed in order for Mews to process th
         {
           "code": "1",
           "cost": 20,
+          "amount": 
+          {
```
Comment (patrikmiksovsky95) [inline mews-api.md]:
```suggestion
          "amount": {
```

File: mews-api.md (hunk 3 above)
Comment (patrikmiksovsky95) [inline mews-api.md]:
```suggestion
  "totalAmount": {
```

File: mews-api.md [hunk 6]
```diff
@@ -813,13 +850,23 @@ _¹ It is required that the code remains the same within each booking modificati
...
 | `pricing` | `int` | required | [`Extra pricing Type`](mews-api.md#extra-pricing-types) code of the extra product pricing. |
 
+#### Amount
+
+| Property | Type |  | Description |
+| --- | --- | --- | --- |
+| `net` | `decimal` | optional | Price with taxes excluded. |
+| `gross` | `decimal` | optional | Price with taxes included. |
+
+
```
Comment (patrikmiksovsky95) [inline mews-api.md]:
```suggestion
```

File: mews-api.md [hunk 7]
```diff
@@ -563,9 +563,9 @@ There are certain rules that need to be followed in order for Mews to process th
...
   * Each reservation should have a unique code within the group. The same code for the reservation should be provided in any following modification message.
   * Each reservation in the group can have different `start`, `end`, `spaceTypeCode`, `ratePlanCode`.
-* Group total cost `group`.`totalCost` is the sum of each `reservation`.`totalCost`, which is the sum of all night costs and total costs of all `extras` of the `reservation`.
-  * All costs in the message are inclusive of VAT.
-  * If for any reason the `group`.`totalCost` is be different, Mews will automatically distribute the missing/additional cost to the nights, so the `group`.`totalCost` is achieved.
+* Group total cost `group`.`totalAmount` is the sum of each `reservation`.`totalAmount`, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.
+  * If for any reason the `group`.`totalAmount` is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.
+  * A reservation `group` amounts should be either `gross` or `net`. Either one value can be sent.
```
Comment (patrikmiksovsky95) [inline mews-api.md]:
```suggestion
  * A reservation `group` amounts should be either `gross` or `net`. At least one value has to be sent. 
```

File: mews-api.md [hunk 8]
```diff
@@ -563,9 +563,9 @@ There are certain rules that need to be followed in order for Mews to process th
   * The whole group is uniquely identified by `channelManagerId` in Mews and in the channel manager extranet.
   * Each reservation should have a unique code within the group. The same code for the reservation should be provided in any following modification message.
   * Each reservation in the group can have different `start`, `end`, `spaceTypeCode`, `ratePlanCode`.
-* Group total cost `group`.`totalCost` is the sum of each `reservation`.`totalCost`, which is the sum of all night costs and total costs of all `extras` of the `reservation`.
-  * All costs in the message are inclusive of VAT.
-  * If for any reason the `group`.`totalCost` is be different, Mews will automatically distribute the missing/additional cost to the nights, so the `group`.`totalCost` is achieved.
+* Group total cost `group`.`totalAmount` is the sum of each `reservation`.`totalAmount`, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.
```
Comment (allisonchihak) [inline mews-api.md]: - Replace 1st phrase with,
"`group`.`totalAmount` is the sum of all `reservation`.`totalAmount` objects in the group,".

File: mews-api.md [hunk 9]
```diff
@@ -563,9 +563,9 @@ There are certain rules that need to be followed in order for Mews to process th
...
+  * If for any reason the `group`.`totalAmount` is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.
```
Comment (allisonchihak) [inline mews-api.md]: Replace with,
"If for any reason the sum of the `reservation`.`totalAmount` is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved."

File: mews-api.md [hunk 10]
```diff
@@ -563,9 +563,9 @@ There are certain rules that need to be followed in order for Mews to process th
...
+  * A reservation `group` amounts should be either `gross` or `net`. At least one value has to be sent.
```
Comment (allisonchihak) [inline mews-api.md]: Remove "a", "reservation `group` amounts" is plural.

File: mews-api.md [hunk 11]
```diff
@@ -793,11 +834,13 @@ There are certain rules that need to be followed in order for Mews to process th
...
 | `to` | `string` | required \(exc. Cancellation\) | End date in format `"yyyy-MM-dd"` \(e.g. `"2017-12-31"` for New Years Eve\). |
-| `totalCost` | `decimal` | required \(exc. Cancellation\) | Total cost of the reservation. |
+| ~~`totalCost`~~ | ~~`decimal`~~ | ~~required \(exc. Cancellation\)~~ | ~~Total cost of the reservation.~~ |
+| `totalAmount` | [`Amount`](mews-api.md#amount) object | required \(exc. Cancellation\) | Total amount of the reservation. |
 | `adultCount` | `int` | required \(exc. Cancellation\) | Number of adults per the reservation. |
 | `childCount` | `int` | optional \(exc. Cancellation\) | Number of children per the reservation. |
 | `state` | `int` | optional | [Reservation State](mews-api.md#reservation-states) code of reservation state. _If not provided, Mews will handle the reservation as _`Created`_ or _`Modified`_._ |
 | `prices` | `decimal` collection | required \(exc. Cancellation\) | Collection of prices for each night of the reservation. _The count of prices in this collection has to correspond with number of nights in the reservation._ |
```
Comment (allisonchihak) [inline mews-api.md]: Isn't this depreciated?

File: mews-api.md [hunk 12]
```diff
@@ -793,11 +834,13 @@ There are certain rules that need to be followed in order for Mews to process th
...
+| `amounts` | [`Amount`](mews-api.md#amount) collection | required \(exc. Cancellation\) | Collection of amounts for each night of the reservation. _The count of amounts in this collection has to correspond with number of nights in the reservation._ |
 | `extras` | [`Extra`](mews-api.md#extra) collection | optional | Collection of extra ordered products for the reservation \(e.g. Breakfast\). _Their total price is included in the _`totalCost`_ of the reservation._ |
```
Comment (allisonchihak) [inline mews-api.md]: Needs to refer to _`totalAmount`_.

File: mews-api.md [hunk 13]
```diff
@@ -813,13 +856,23 @@ _¹ It is required that the code remains the same within each booking modificati
 
 #### Extra
 
+* Total cost of the extra product should be sent in `net` or `gross` amounts. Either both `gross` and `net` amounts, or one of them should be sent.
```
Comment (allisonchihak) [inline mews-api.md]: This says sending `gross`, `net`, or both are supported. Why is this different from line 568?

File: mews-api.md [hunk 14]
```diff
@@ -563,9 +563,9 @@ There are certain rules that need to be followed in order for Mews to process th
...
   * Each reservation should have a unique code within the group. The same code for the reservation should be provided in any following modification message.
   * Each reservation in the group can have different `start`, `end`, `spaceTypeCode`, `ratePlanCode`.
-* Group total cost `group`.`totalAmount` is the sum of each `reservation`.`totalAmount`, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.
-  * If for any reason the `group`.`totalAmount` is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.
-  * A reservation `group` amounts should be either `gross` or `net`. At least one value has to be sent. 
+* Group total cost `group`.`totalAmount` is the sum of all `reservation`.`totalAmount` objects in the group, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.
+  * If for any reason the sum of the `reservation`.`totalAmount` is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.
```
Comment (allisonchihak) [inline mews-api.md]:
```suggestion
  * If for any reason the sum of the `reservation`.`totalAmount` objects is different, Mews will automatically distribute the missing/additional amount to the nights, so the `group`.`totalAmount` is achieved.
```

File: mews-api.md [hunk 15]
```diff
@@ -401,7 +401,7 @@ This is example of a _successful_ response. In case an error occurred, the respo
 
 | Property | Type |  | Description |
 | --- | --- | --- | --- |
-| `amount` | `decimal` | required | Defines the amount of the absolute fee. |
+| `amount` | `decimal` | required | Defines the amount of the absolute fee. Sent in `gross` |
```
Comment (siroky) [inline mews-api.md]: No dot at the end of sentence.

File: mews-api.md [hunk 16]
```diff
@@ -563,9 +563,9 @@ There are certain rules that need to be followed in order for Mews to process th
   * The whole group is uniquely identified by `channelManagerId` in Mews and in the channel manager extranet.
   * Each reservation should have a unique code within the group. The same code for the reservation should be provided in any following modification message.
   * Each reservation in the group can have different `start`, `end`, `spaceTypeCode`, `ratePlanCode`.
-* Group total cost `group`.`totalCost` is the sum of each `reservation`.`totalCost`, which is the sum of all night costs and total costs of all `extras` of the `reservation`.
-  * All costs in the message are inclusive of VAT.
-  * If for any reason the `group`.`totalCost` is be different, Mews will automatically distribute the missing/additional cost to the nights, so the `group`.`totalCost` is achieved.
+* Group total cost `group`.`totalAmount` is the sum of all `reservation`.`totalAmount` objects in the group, which is the sum of all night amounts and total amounts of all `extras` of the `reservation`.
```
Comment (siroky) [inline mews-api.md]: I'd make dot part of the code fragment...

File: mews-api.md [hunk 17]
```diff
@@ -641,6 +641,10 @@ There are certain rules that need to be followed in order for Mews to process th
         {
           "code": "1",
           "cost": 20,
```
Comment (siroky) [inline mews-api.md]: Shouldn't we remove all the obsolete things from the examples and also from the documentation?

File: mews-api.md [hunk 18]
```diff
@@ -712,7 +752,8 @@ There are certain rules that need to be followed in order for Mews to process th
 | `channelId` | `string` | required \(always\) | Unique identification of the booking in the Channel \(i.e. OTA\). |
 | `channelManagerId` | `string` | required \(always\) | Unique identification of the booking in the channel manager. |
 | `currencyCode` | `string` | required \(exc. Cancellation\) | 3 letter code of currency of any price within the booking. |
-| `totalCost` | `decimal` | required \(exc. Cancellation\) | Total cost of the whole booking. |
+| ~~`totalCost`~~ | ~~`decimal`~~ | ~~required \(exc. Cancellation\)~~ | ~~Total cost of the whole booking.~~ |
```
Comment (siroky) [inline mews-api.md]: This I'd rather fully remove, just like we do it in Connector.

File: mews-api.md [hunk 19]
```diff
@@ -685,11 +703,29 @@ There are certain rules that need to be followed in order for Mews to process th
         120,
         120
       ],
+      "amounts": [
```
Comment (siroky) [inline mews-api.md]: Consider using the same formatting as in Connector API - meaning 4 space indentation. Especially with very nested examples, it's much more readable IMHO.

File: mews-api.md (hunk 17 above)
Comment (patrikmiksovsky95) [reply mews-api.md]: @siroky I left them there until we remove them completely we have issue with deadline to remove them completely

File: mews-api.md (hunk 16 above)
Comment (allisonchihak) [reply mews-api.md]: Do you mean the dots between `group`.`totalCost` and `reservation`.`totalAmount` or the dot after `reservation`?

File: mews-api.md (hunk 17 above)
Comment (siroky) [reply mews-api.md]: I know, usually in connector, we delete from documentation immediately, but support in code until deadline. Discuss with @vitasamek

**Pull request merged**